### PR TITLE
Update Spring Boot to version 2.4.13

### PR DESCRIPTION
**Pull request by matthew-morris-cessda**
This PR migrates CVS from the Elasticsearch Transport client to the Elasticsearch REST client. This is necessary because of depencency updates bundled with Spring Boot 2.4.13.

The supported versions of Elasticsearch has been raised to version 7.9.3.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -721,11 +712,12 @@
             <plugin>
                 <groupId>com.github.alexcojocaru</groupId>
                 <artifactId>elasticsearch-maven-plugin</artifactId>
-                <version>6.28</version>
+                <version>6.26</version>
```
[the remark this answers is not in the document]
Comment (matthew-morris-cessda) [reply pom.xml:715]: That's a mistake. I'll correct this.

File: src/main/java/eu/cessda/cvs/web/rest/FileUploadResource.java
Comment (alen-vodopijevec-cessda) [inline src/main/java/eu/cessda/cvs/web/rest/FileUploadResource.java:1]: @matthew-morris-cessda .. what do you think about this XSS warning for the user-provided value?

Comment (matthew-morris-cessda) [reply src/main/java/eu/cessda/cvs/web/rest/FileUploadResource.java:1]: It probably should be swapped for a generated value, but not in this PR

File: src/main/resources/config/application-prod.yml [hunk 2]
```diff
@@ -50,14 +47,9 @@ spring:
...
-    elasticsearch:
-      cluster-name: docker-cluster
-      cluster-nodes: cvs-elasticsearch:9300
-      properties:
-        path:
-          logs: target/elasticsearch/log
-          data: target/elasticsearch/data
+  elasticsearch:
+    rest:
+      uris: http://localhost:9200
```
Comment (alen-vodopijevec-cessda) [inline src/main/resources/config/application-prod.yml:52]: Port number change is correct?

File: src/main/resources/config/application-dev.yml [hunk 3]
```diff
@@ -61,15 +59,9 @@ spring:
...
-      cluster-name: docker-cluster
-      cluster-nodes: 127.0.0.1:9300
-      properties:
-        path:
-          logs: target/elasticsearch/log
-          data: target/elasticsearch/data
-          home: target/elasticsearch
+  elasticsearch:
+    rest:
+      uris: http://localhost:9200
```
Comment (alen-vodopijevec-cessda) [inline src/main/resources/config/application-dev.yml:64]: Port number change?

File: src/main/resources/config/application-dev.yml [hunk 4]
```diff
@@ -42,7 +40,7 @@ spring:
       indent-output: true
   datasource:
     type: com.zaxxer.hikari.HikariDataSource
-    url: jdbc:mysql://localhost:3306/cvs?useUnicode=true&characterEncoding=utf8&useSSL=false&useLegacyDatetimeCode=false&serverTimezone=UTC&createDatabaseIfNotExist=true
+    url: jdbc:mysql://localhost:3311/cvs?useUnicode=true&characterEncoding=utf8&useSSL=false&useLegacyDatetimeCode=false&serverTimezone=UTC&createDatabaseIfNotExist=true
```
Comment (alen-vodopijevec-cessda) [inline src/main/resources/config/application-dev.yml:43]: Port number change?

File: src/test/resources/config/application.yml [hunk 5]
```diff
@@ -43,15 +43,9 @@ spring:
...
-      cluster-name: docker-cluster
-      cluster-nodes: 127.0.0.1:9300
-      properties:
-        path:
-          logs: target/elasticsearch/log
-          data: target/elasticsearch/data
-          home: target/elasticsearch
+  elasticsearch:
+    rest:
+      uris: http://localhost:9225
```
Comment (alen-vodopijevec-cessda) [inline src/test/resources/config/application.yml:48]: Port number change

Comment (matthew-morris-cessda) [reply src/test/resources/config/application.yml:48]: Elasticsearch now uses the HTTP port, this is correct

File: src/main/resources/config/application-dev.yml (hunk 3 above)
Comment (matthew-morris-cessda) [reply src/main/resources/config/application-dev.yml:64]: This is correct

File: src/main/resources/config/application-prod.yml (hunk 2 above)
Comment (matthew-morris-cessda) [reply src/main/resources/config/application-prod.yml:52]: This is correct

**Review by alen-vodopijevec-cessda — APPROVED** [review]

**Pull request merged**